### PR TITLE
511 modulenotfounderror no module named mapieconformity scoresresidual conformity scores

### LINKED ISSUE #511: ModuleNotFoundError: No module named 'mapie.conformity_scores.residual_conformity_scores'
Hello MAPIE team
Been happily using MAPIE for a while now. Filling this because I'm not sure if the regression is intended from the `0.9.0` release or not.

**Describe the bug**
Failing to load previously trained MAPIE models

```python
ModuleNotFoundError: No module named 'mapie.conformity_scores.residual_conformity_scores'
 ```

**To Reproduce**
- Trained MapieRegressor instanced over LinearRegression and RandomForest using `MAPIE==0.8.4`
- Pickled them with joblib
- Loading them to memory again with `MAPIE < 0.9.0` (up to 0.8.6) works fine 
- With MAPIE 0.9.0 I get this error: ModuleNotFoundError: No module named 'mapie.conformity_scores.residual_conformity_scores' 

**Expected behavior**

**Screenshots**

**Desktop (please complete the following information):**
 - OS: Ubuntu on WSL Windows 11
 - Browser Firefox
 - MAPIE Version work with 0.8.6, breaks with 0.9.0
 - Python version: tested with 3.12.5

**Additional context**

### PR DESCRIPTION
**Pull request by vincentblot28**
Fixes #511

### CODE REVIEW

**Comment by Valentin-Laurent** [comment]
Hello @vincentblot28,
Conformity scores are not the only imports breaking with MAPIE 0.9.0, I also listed the following:
```
from mapie.estimator.estimator import EnsembleRegressor
from mapie.conformity_scores.utils_classification_conformity_scores import get_true_label_position
```

Finally, `theoretical_description_conformity_scores.rst` now mentions outdated class names.

Let me know if you can take care of the above points. Thanks!

File: mapie/conformity_scores/conformity_scores.py
Comment (Valentin-Laurent) [inline mapie/conformity_scores/conformity_scores.py:1]: Do we have to formally deprecate the class `ConformityScore`, as it is used internally only? Also, this class doesn't exist anymore in v0.9.0 if I'm not mistaken, so the warning should mention the new relevant classes.

File: mapie/conformity_scores/residual_conformity_scores.py
Comment (Valentin-Laurent) [inline mapie/conformity_scores/residual_conformity_scores.py:1]: I liked your solution with imports only. Typing checker was not happy with that, is that it?

File: mapie/conformity_scores/conformity_scores.py
Comment (vincentblot28) [reply mapie/conformity_scores/conformity_scores.py:1]: my bad there is a typo here, the warning message should be 
 "WARNING: Deprecated path to import ConformityScore. "
    "Please prefer the new path: "
    "[from mapie.conformity_scores import BaseRegressionScore]."
    
    If some user had custom conformity score, we have to keep the compatibility

**Review by thibaultcordier — APPROVED** [review]
Great, thanks for also handling the other classes that were likely to have the same problem.

**Pull request merged**